src/java/picard/analysis/CollectMultipleMetrics.java

```diff
@@ -325,27 +325,29 @@ public SinglePassSamProgram makeInstance(final String outbase, final File input,
     private final Set<MetricAccumulationLevel> accumLevelDefault = CollectionUtil.makeSet(MetricAccumulationLevel.ALL_READS);
 
     @Option(shortName="LEVEL", doc="The level(s) at which to accumulate metrics.")
-    public Set<MetricAccumulationLevel> METRIC_ACCUMULATION_LEVEL = new HashSet<MetricAccumulationLevel>(accumLevelDefault);
+    public Set<MetricAccumulationLevel> METRIC_ACCUMULATION_LEVEL = new HashSet<>(accumLevelDefault);
 
     @Option(shortName = "EXT", doc="Append the given file extension to all metric file names (ex. OUTPUT.insert_size_metrics.EXT). None if null", optional=true)
     public String FILE_EXTENSION = null;
 
-    @Option(doc = "List of metrics programs to apply during the pass through the SAM file.")
-    public List<Program> PROGRAM = CollectionUtil.makeList(Program.CollectAlignmentSummaryMetrics, Program.CollectBaseDistributionByCycle,
+    @Option(doc = "Set of metrics programs to apply during the pass through the SAM file.")
+    public Set<Program> PROGRAM = CollectionUtil.makeSet(Program.CollectAlignmentSummaryMetrics, Program.CollectBaseDistributionByCycle,
```

Comment:
I see. I understand now. Will do.

On Tue, Apr 19, 2016 at 6:17 AM, Tim Fennell <notifications@github.com>
wrote:

> In src/java/picard/analysis/CollectMultipleMetrics.java
> <https://github.com/broadinstitute/picard/pull/479#discussion_r60206062>:
>
> >
> >      @Option(shortName = "EXT", doc="Append the given file extension to all metric file names (ex. OUTPUT.insert_size_metrics.EXT). None if null", optional=true)
> >      public String FILE_EXTENSION = null;
> >
> > -    @Option(doc = "List of metrics programs to apply during the pass through the SAM file.")
> > -    public List<Program> PROGRAM = CollectionUtil.makeList(Program.CollectAlignmentSummaryMetrics, Program.CollectBaseDistributionByCycle,
> > +    @Option(doc = "Set of metrics programs to apply during the pass through the SAM file.")
> > +    public Set<Program> PROGRAM = CollectionUtil.makeSet(Program.CollectAlignmentSummaryMetrics, Program.CollectBaseDistributionByCycle,
>
> When CommandLineProgram emits the command line (both to stderr, and/or in
> any metrics file headers) it *re*constructs it from the @Option instance
> members.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/479/files/1469c9c29fca92855127826ed588ea72f57d07f2#r60206062>
>